### PR TITLE
Markdown support

### LINKED ISSUE #258: Markdown support of announcements is missing
Regression from #218

### PR DESCRIPTION
**Pull request by nickvergessen**
Fix #258

### CODE REVIEW

**Comment by nickvergessen** [comment]
cc @splitt3r you want to test this?

**Comment by nickvergessen** [comment]
@sahalsaad @schmuuu care to review since you helped with the previous implementation: https://github.com/nextcloud/announcementcenter/pull/130

**Comment by splitt3r** [comment]
I will try to find the time to have a look at it 👍🏻. But i can't promise anything as i'm not using the app and have to setup the dev environment first.

File: package.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
 		"stylelint:fix": "stylelint css/*.css css/*.scss src/**/*.scss src/**/*.vue --fix"
 	},
 	"dependencies": {
+		"@juliushaertl/vue-richtext": "^1.0.1",
```
Comment (splitt3r) [inline package.json:17]: Any reason for not moving this under the Nextcloud org?

Comment (nickvergessen) [reply package.json:17]: Feel free to poke here https://github.com/juliushaertl/vue-richtext/issues :D

**Comment by Schmuuu** [comment]
Hi @nickvergessen, 
I'm sorry for the late reply, have to work a lot. Could you let me know, how I can install this version for testing? Do I simply have to download this branch (https://github.com/nextcloud/announcementcenter/tree/bugfix/258/markdown-support) as zip file and extract it into nextcloud/apps/announcementcenter or what are the exact steps?

**Comment by splitt3r** [comment]
> 
> 
> Hi @nickvergessen,
> I'm sorry for the late reply, have to work a lot. Could you let me know, how I can install this version for testing? Do I simply have to download this branch (https://github.com/nextcloud/announcementcenter/tree/bugfix/258/markdown-support) as zip file and extract it into nextcloud/apps/announcementcenter or what are the exact steps?

Yes

**Comment by nickvergessen** [comment]
Well you need to build the js/ folder too. I will upload a package tomorrow

**Comment by Schmuuu** [comment]
A few seconds before your comment I downloaded the zip and extracted it into the apps folder. I had to click the "Continue" button in the web browser for the update of this app.

The final message however was: "The following apps have been disabled: announcementcenter (incompatible)"
I noticed that it requires NC 21. Is it possible to run this app on NC 20.0.x as well? I could manually change the min version in the info.xml if you say, the app should run in NC20 as well.

**Comment by splitt3r** [comment]
image

External mages won´t work by intention right? Maybe a small hint that external images won't work?

**Comment by nickvergessen** [comment]
> External mages won´t work by intention right? Maybe a small hint about it? Could be very interesting vor announcements. Sth. like an marketing image from the companies website.

Well same URL should work, it's the CSP issue. Maybe we add a allow-list option in the admin settings at some point. But at least it's unchanged from old v4 behaviour.

> What about a rich(er) text editor? With buttons for bold etc.

It's a future topic, since it's not a regression from v4 at least. I plan to use https://nextcloud-vue-components.netlify.app/#/Components/RichContenteditable Doesn't have any buttons yet, but at least would allow mentions.

**Comment by nickvergessen** [comment]
Thanks for testing and the feedback!

**Comment by nickvergessen** [comment]
Copied the feedback to https://github.com/nextcloud/announcementcenter/issues/283 and https://github.com/nextcloud/announcementcenter/issues/284

**Pull request merged**